perception/traffic_light_arbiter/src/signal_match_validator.cpp

```diff
@@ -0,0 +1,274 @@
+// Copyright 2024 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "traffic_light_arbiter/signal_match_validator.hpp"
+
+namespace util
+{
+using TrafficSignalArray = autoware_perception_msgs::msg::TrafficSignalArray;
+using TrafficSignal = autoware_perception_msgs::msg::TrafficSignal;
+using Element = autoware_perception_msgs::msg::TrafficSignalElement;
+using Time = builtin_interfaces::msg::Time;
+
+// Finds a signal by its ID within a TrafficSignalArray
+std::optional<TrafficSignal> find_signal_by_id(
+  const TrafficSignalArray & signals, int64_t signal_id)
+{
+  auto it = std::find_if(
```

Comment:
nit: This makes a O(n) search. Would it be better to use unordered maps?

Reply:
Addressed in https://github.com/autowarefoundation/autoware.universe/pull/6423/commits/95fd4dc754e6790a07efec5da2e597c26de8b347